python/BioSimSpace/_SireWrappers/_molecule.py

```diff
@@ -1269,10 +1273,24 @@ def _toPertFile(self, filename="MORPH.pert", zero_dummy_dihedrals=False,
 
         # Seed the random number generator so that we get reproducible atom names.
         # This is helpful when debugging since we can directly compare pert files.
-        _random.seed(42)
+        _random.seed(42)        
+
+        if pert_type:
+          if type(pert_type) is not str:
+              raise TypeError("'pert_type' must be of type 'str'")
+
+          allowed_pert_types = [  
+                                "standard",
+                                "1_discharge_soft",
+                                "2_vanish_soft",
+                                "3_flip",
+                                "4_grow_soft",
+                                "5_charge_soft"]
```

Comment:
With regards to the numbering: Am I right in remembering that the steps need to be performed in a specific order? If not, it might be better to remove numbering. Also, is it possible to conceive of other perturbation types that could be added in addition to these, e.g. as other steps in between? If so, then we'd need to renumber if we did add support for other perturbation types, which would break backwards compatibility.

Reply:
I added the numbering at the time to make the steps somewhat more intuitive (to me, at least). In principle the steps are not dependent on each other in any way, but users will have to run all five of them to get meaningful free energies (by summation).

IIRC `3_flip` consists of multiple actions, i.e. perturbing bonds, angles and all hard atom parameters. If ever we would want to break that up into two steps then yes, we would have to renumber the steps in that case, but I'm not sure why we would want to do that (famous last words)

Reply:
okay, I'll remove the numbering.

Reply:
I'll also adjust `standard` to `full`, I agree this is clearer.

Reply:
see 44cce8823acde7e44a5bf5defdbe1dd822973815 and 2fa10bb54f3315a5d2cf7a44d16224e51d2392d4